pkg/tcpping/tcpping.go

```diff
@@ -0,0 +1,35 @@
+package tcpping
+
+import (
+	"net"
+	"time"
+)
+
+// TCPPinger repeatedly attempts to open a TCP connection
+type TCPPinger struct {
```

Comment:
The "er" is pretty much "reserved" (with exceptions, even in the standard libs) for interfaces, like `io.Reader`. I would not name it Ping*er*

Reply:
Do you have a suggestion? In Java I would call it `TCPPingPortWaiter` because it implements the `PortWaiter` interface by doing a TCP Ping, but that doesn't seem like go style.

Reply:
I would not be aware of any "rules". Maybe my first choice for a concrete implementation would be the same name without a "er". But certainly not a rule.

Giving it a second thought I would not make this an interface at all. Why is it one? Mocking/testing? I mean how many implementations do we expect? In an internal package. Exactly 1, so why make it an interface in the first place? It is just one concrete helper/utility.

Reply:
It's for mocking/testing and general modularization. The implementation uses time and TCP/IP, which are both a pain for tests. Time especially so.

Reply:
It's for mocking/testing and general modularization. The implementation uses time and TCP/IP, which are both a pain for tests. Time especially so.

Reply:
nah, "modularization" for something that will never see a real 2nd implementation IMO does not improve anything, it just adds a dead interface. But I totally see the point why it eases testing here. So let's just get over it and merge it :)